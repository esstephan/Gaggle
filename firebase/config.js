import firebase from 'firebase';

let config = {
  apiKey: "",
  authDomain: "",
  databaseURL: "",
  storageBucket: "",
  messagingSenderId: ""
};

export default firebaseRef = firebase.initializeApp(config);