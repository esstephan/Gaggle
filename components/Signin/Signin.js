import React, { Component } from 'react';
import { Container, Header, Footer, Content, Form, Item, Input, Icon, Button, FooterTab, Text } from 'native-base';
import { firebaseRef, firebaseDB } from '../../firebase/firebaseHelpers';
import GroupView from './../../components/GroupView/GroupView';

const styles = {
  marginBottom: {
    marginBottom: 10,
  },
};

export default class Signin extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      showSignUp: true,
    };

    firebaseRef.auth().onAuthStateChanged((user) => {
      if (user) {
        firebaseDB.ref(`users/${user.uid}`).once('value').then((snapshot) => {
          this.handleChangePage(snapshot.val());
          console.log(snapshot.val());
        });
      }
    });

    this.handleChangePage = this.handleChangePage.bind(this);
    this.signup = this.signup.bind(this);
    this.signin = this.signin.bind(this);
    this.logout = this.logout.bind(this);
  }

  handleChangePage(user) {
    this.props.navigator.push({
      component: GroupView,
      title: 'Group Members',
      leftButtonTitle: 'Log Out',
      onLeftButtonPress: () => {
        this.logout();
        this.props.navigator.pop();
      },
      passProps: {
        user,
      },
    });
  }


  signup() {
    firebaseRef.auth().createUserWithEmailAndPassword(this.state.email, this.state.password)
    .then(() => {
      const user = firebaseRef.auth().currentUser;
      const displayName = user.email.split('@')[0];

      user.updateProfile({
        displayName,
      })
      .then(() => {
        const newUserObj = {
          displayName: user.displayName,
          email: user.email,
          uid: user.uid,
          // Location may need to be revisited or handled wherever it is set/read
          location: {},
        };

        firebaseDB.ref(`users/${user.uid}`).set(newUserObj).then((snapshot) => {
          this.handleChangePage(snapshot.val());
        });
        console.log('Name set up successful!');
      }, (error) => {
        console.log('Name set up unsuccessful', error);
      });
    })
    .catch((error) => { console.log(`Error ${error}`); });
  }

  signin() {
    firebaseRef.auth().signInWithEmailAndPassword(this.state.email, this.state.password)
    .catch((error) => { console.log(`Error ${error}`); });
  }

  logout() {
    firebaseRef.auth().signOut().then(() => {
      console.log('Sign-out Successful.');
    }, (error) => {
      console.log('Sign-out failed.', error);
    });
  }

  render() {
    return (
      <Container >
        <Header />
        <Content style={{ padding: 10 }}>
          <Form>
            <Item style={styles.marginBottom} regular>
              <Input
                onChangeText={text => this.setState({ email: text })}
                placeholder="Email"
                autoCapitalize="none"
              />
              {/.+@.+\..+/i.test(this.state.email) && <Icon name={'checkmark-circle'} style={{ color: 'green' }} />}
            </Item>
            <Item regular>
              <Input
                onChangeText={(text) => { this.setState({ password: text }); }}
                placeholder="Password"
                autoCapitalize="none"
                secureTextEntry
              />
              {this.state.password.length >= 6 && <Icon name={'checkmark-circle'} style={{ color: 'green' }} />}
            </Item>
          </Form>

          {this.state.showSignUp ?
            <Button
              style={{ padding: 5 }}
              onPress={() => this.setState({ showSignUp: false })}
              transparent
            >
              <Text>Already registered?</Text>
            </Button> :
            <Button
              style={{ padding: 5 }}
              onPress={() => this.setState({ showSignUp: true })}
              transparent
            >
              <Text>{'Don\'t have an account?'}</Text>
            </Button>
          }

          {this.state.showSignUp ?
            <Button disabled={this.state.password.length < 6} onPress={this.signup}>
              <Text>Sign up</Text>
            </Button> :
            <Button disabled={this.state.password.length < 6} onPress={this.signin}>
              <Text>Sign in</Text>
            </Button>
          }

        </Content>
        <Footer>
          <FooterTab>
            <Button onPress={this.handleChangePage}>
              <Text>Next Page</Text>
            </Button>
          </FooterTab>
        </Footer>
      </Container>
    );
  }
}

Signin.propTypes = {
  navigator: React.PropTypes.object.isRequired,
};
