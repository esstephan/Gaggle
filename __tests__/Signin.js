import 'react-native';
import React from 'react';
import Signin from '../components/Signin/Signin.js';
import { firebaseRef } from '../firebase/config.js';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Signin />
  );
  expect(tree).toMatchSnapshot();

});

