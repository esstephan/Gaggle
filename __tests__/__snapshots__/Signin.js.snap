exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "top": 0,
        }
      }
      virtual={undefined} />
  </View>
  <RCTScrollView>
    <View>
      <View
        style={Object {}}
        virtual={undefined}>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "flexDirection": "row",
              "marginLeft": 0,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}>
          <TextInput
            autoCapitalize="none"
            editable={true}
            onChangeText={[Function]}
            placeholder="Username"
            placeholderTextColor="#575757"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "lineHeight": 24,
                "paddingLeft": 8,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            virtual={undefined} />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "flexDirection": "row",
              "marginLeft": 0,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}>
          <TextInput
            autoCapitalize="none"
            editable={true}
            onChangeText={[Function]}
            placeholder="6 or more characters"
            placeholderTextColor="#575757"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "lineHeight": 24,
                "paddingLeft": 8,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            virtual={undefined} />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#b5b5b5",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Object {
              "alignSelf": "center",
              "zIndex": 999,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          Sign up
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#b5b5b5",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Object {
              "alignSelf": "center",
              "zIndex": 999,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          Sign in
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderColor": "#cbcbcb",
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "left": 0,
        "right": 0,
      }
    }
    virtual={undefined}>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
      virtual={undefined}>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": null,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Object {
              "alignSelf": "center",
              "zIndex": 999,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          Next Page
        </Text>
      </View>
    </View>
  </View>
</View>
`;
